tidy3d/components/data/monitor_data_n2f.py

```diff
@@ -36,24 +90,78 @@ def normalize(self, source_spectrum_fn: Callable[[float], complex]) -> AbstractN
 
         return self.copy(update=fields_norm)
 
-    @staticmethod
-    def nk(frequency: float, medium: Medium) -> Tuple[float, float]:
+    def nk(self, medium: Medium) -> Tuple[float, float]:
         """Returns the real and imaginary parts of the background medium's refractive index."""
-        eps_complex = medium.eps_model(frequency)
-        return medium.eps_complex_to_nk(eps_complex)
+        return medium.nk_model(frequency=self.f)
 
     @staticmethod
-    def k(frequency: float, medium: Medium) -> complex:
-        """Returns the complex wave number associated with the background medium."""
-        index_n, index_k = AbstractNear2FarData.nk(frequency, medium)
+    def propagation_factor(frequency: float, medium: Medium) -> complex:
+        """Complex valued wavenumber associated with a frequency and medium."""
+        index_n, index_k = medium.nk_model(frequency=frequency)
         return (2 * np.pi * frequency / C_0) * (index_n + 1j * index_k)
 
-    @staticmethod
-    def eta(frequency: float, medium: Medium) -> complex:
+    def k(self, medium: Medium) -> complex:
+        """Returns the complex wave number associated with the background medium."""
+        return self.propagation_factor(frequency=self.f, medium=medium)
+
+    def eta(self, medium: Medium) -> complex:
         """Returns the complex wave impedance associated with the background medium."""
-        eps_complex = medium.eps_model(frequency)
+        eps_complex = medium.eps_model(frequency=self.f)
         return ETA_0 / np.sqrt(eps_complex)
 
+    def near_field_rad_vecs(self, medium: Medium) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
I just noticed this: I'm not sure it's correct to call these `near_field` - all the field quantities that show up in this file (rad vecs, fields) are the result of projecting to the observation grid and making those various approximations during the integration, and so would be considered far field quantities, I think. Maybe this could be called `rad_vecs_to_fields`?